### DISCUSSION TITLE
updated hook not triggered after property has been updated

### DISCUSSION THREAD

**Discussion by maltebaer** [livewire/livewire#7905] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.4

### Laravel version

v10.43.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. change/update property `selectedGreekLetter` (works)
2. expect property `letter` to be updated as well (fails)

```php
<?php
namespace App\Livewire;

use Livewire\Component;

class Simple extends Component
{
    public array $greekAlphabet = ['alpha', 'beta', 'gamma'];

    public ?string $selectedGreekLetter = null;

    public ?string $letter = null;

    public function updated(): void
    {
        logs()->info('updated');
    }

    public function updatedSelectedGreekLetter(): void
    {
        $this->letter = match ($this->selectedGreekLetter) {
            'alpha' => 'A',
            'beta' => 'B',
            'gamma' => 'C',
            default => null,
        };
    }

    public function selectGreekLetter(string $greekLetter): void
    {
        $this->selectedGreekLetter = $greekLetter;
    }
}
```

```html
<div>
    <div class="flex gap-2 text-teal-500">
        @foreach ($greekAlphabet as $greekLetter)
            <button
                wire:click="selectGreekLetter('{{ $greekLetter }}')"
                @class(['underline' => $greekLetter === $selectedGreekLetter])
            >
                {{ $greekLetter }}
            </button>
        @endforeach
    </div>

    <div class="text-rose-500">@json($letter)</div>
</div>
```

### Repository or Wirebox URL

https://wirebox.app/b/y828d

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
i think it's only works only with `wire:model` in client side (not with `wire:click`).
so you can throw your logic in the clicked function not in updated property.

> Client-side users can update public properties in many different ways, 
> most commonly by modifying an input with `wire:model` on it.
> https://livewire.laravel.com/docs/lifecycle-hooks#update

i use radio input for the example:
https://wirebox.app/b/yrdo3

**Reply by maltebaer** [reply]
thanks for your help! however, i'm not entirely convinced that `wire:model` would be the right choice.

i am actually trying to update the value of a child component using events. the update works, but does not trigger the `updated` hook.

i've modified the wirebox example: https://wirebox.app/b/y828d

i'm confused and not sure if this is a bug or if i'm misunderstanding something.

**Reply by joshhanley** [reply, marked as answer]
@maltebaer the updated hooks are only triggered when the change comes from the front end. As you're changing the property on the backend, Livewire has no way to detect that to trigger the updated hook.

What I would do is extract the logic out of the updated hook and into a dedicated method, that you can call from within the updated hook and also call it from your `selectGreekLetter()` method.

Hope this helps!

**Reply by maltebaer** [reply]
oh, that totally makes sense! i already "solved" the issue by moving the logic into a dedicated method, but was so confused about why the `updated` hook was not triggered.

thanks a lot for clarifying!

**Reply by joshhanley** [reply]
@maltebaer nice! No worries! 🙂

**Comment by michaelklopf** [comment]
Does anyone have had the problem yet, that setting a value to `null` doesn't trigger the updatedValue hook? 

**edit**
Problem found, was because of not setting value to the default value in mount() and changing from null to null did not trigger an update.
